src/models/HintItem.ts

```diff
@@ -9,6 +9,7 @@ export interface HintItem {
   id: string;
   hint: boolean;
   goalHints: IGoalHint[];
+  deletedGoalHints?: Omit<IGoalHint, "id">[];
```

Comment:
shouldn't we just store id here ?

Reply:
the response we get from the api is an hint goal object with title, parentTitle and duration? fields. We add id to it when we store the hint goal in collection. Thats why used title identify the hint goal

Reply:
@vinaybadgujar102 i was referring to deletedGoalHints field.

Reply:
i was referring to same line only. The backend doesn't respond with id for hintGoal that's why we cant uniquely identify it. Thats why we are comparing with title. The id gets added in frontend only.

@tijlleenders can we change the hints API route to store id of each hint

Reply:
No the API doesn't store hint IDs as they are anonymously shared amongst all users.

Reply:
Once you add a Hint it should get an ID (generated) though - and you can use that normally in the deleted collection.

Reply:
> Once you add a Hint it should get an ID (generated) though - and you can use that normally in the deleted collection.

Add hint where? locally or server?

Reply:
When you add the suggested hint (without uuid) to your local MyGoals collection - it will get an autogenerated uuid, right?

Reply:
yes, id will get added locally. But if we toggle the hints option to off, then we delete the hints array stored locally for that particular goal. When we turn hints back on again, we again add new ids and then store that hints array. So the id gets changed and thus we cant track which hint was deleted if we compare with ids.